torchdata/datapipes/iter/util/converter.py

```diff
@@ -67,6 +68,12 @@ def __getitem__(self, index):
             self._load_map()
         return self._map[index]  # type: ignore[index]
 
+    def __iter__(self):
+        if self._map is None:
+            self._load_map()
+        else:
+            yield from self._map.values()
+
```

Comment:
I am sorry that I got confused by these two PRs. Why do we need `__iter__` for this `MapDataPipe`?

Reply:
It is not needed but without it you can't iterate over it. I consider it to be a nice-to-have.

```python
from torchdata.datapipes.iter import IterableWrapper
from torchdata.datapipes.map import IterToMapConverter

dp = IterableWrapper([(i, i) for i in range(10)])
dp = IterToMapConverter(dp)
for i in iter(dp):
    print(i)
```

In the end, you will get `KeyError: 10.` if `__iter__` isn't there.

Reply:
IIRC, You iterate it if you raise proper Error
```py
def __getitem__(self, index):
    try:
        do your stuff
    except KeyError:
        raise IndexError

```